cacheService.js

```diff
@@ -152,11 +152,17 @@ function cacheService(cacheServiceConfig, cacheModules) {
       expiration = null;
     }
     log(false, '.set() called:', {key: key, value: value});
+    var moduleCallbacks = 0;
     for(var i = 0; i < self.cachesLength; i++){
       var cache = self.caches[i];
       var ref = (i === self.caches.length - 1) ? refresh : null;
-      var func = (i === 0) ? cb : noop;
-      cache.set(key, value, expiration, ref, func);
+      cache.set(key, value, expiration, ref, moduleCallback);
+    }
+    function moduleCallback(err){
+      if (++moduleCallbacks === self.cachesLength){
+        // note: no errors are passed back!
+        cb(null);
+      }
```

Comment:
Thinking about this some more, it should probably just use [async](https://github.com/caolan/async)'s [`map()`](http://caolan.github.io/async/docs.html#map) to make the `.set()` calls on all of the modules and collect the results.  This is exactly the kind of synchronization management for which async was designed.